File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.js
var smallestIndex = function(nums) {
    for (let i = 0; i < nums.length; i++) {
        let digitSum = 0;
        let currentNumber = nums[i];

        while (currentNumber > 0) {
            digitSum += currentNumber % 10;
            currentNumber = Math.floor(currentNumber / 10);
        }

        if (digitSum === i) {
            return i;
        }
    }

    return -1;
};

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int smallestIndex(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            int currentNumber = nums[i];
            int digitSum = 0;

            while (currentNumber > 0) {
                digitSum += currentNumber % 10;
                currentNumber /= 10;
            }

            if (digitSum == i) {
                return i;
            }
        }

        return -1;
    }
}